### PR TITLE
Handle pastEvents

### PR DESCRIPTION
**Pull request by lmeier**

### CODE REVIEW

File: RadarSDK/RadarAPIClient.m [hunk 1]
```diff
@@ -138,6 +138,18 @@ - (void)flushReplays:(NSArray<NSDictionary *> *_Nonnull)replays
...
+                            id eventsObj = res[@"events"];
+
+                            NSArray<RadarEvent *> *events = [RadarEvent eventsFromObject:eventsObj];
+                            if (events.count) {
+                                NSArray *eventsArr = (NSArray *)eventsObj;
+                                NSDictionary *lastEvent = [eventsArr lastObject];
+                                NSDictionary *userObj = lastEvent[@"user"];
+
+                                RadarUser *user = [[RadarUser alloc] initWithObject:userObj];
+                                [[RadarDelegateHolder sharedInstance] didReceiveEvents:events user:user];
```
Comment (corypisano) [inline RadarSDK/RadarAPIClient.m:150]: want to sanity check here if this handles if user is undefined / not passed down. would be good to have the option of just passing through events

Comment (lmeier) [reply RadarSDK/RadarAPIClient.m:150]: In the transition from 3.1 to 3.2, we made this optional: https://github.com/radarlabs/radar-sdk-ios/blob/559948728bcdb78c937e885a2cf33d8eb46836c1/MIGRATION.md#L67

You can see an example in the trip functions, e.g. https://github.com/radarlabs/radar-sdk-ios/blob/559948728bcdb78c937e885a2cf33d8eb46836c1/RadarSDK/RadarAPIClient.m#L619

File: RadarSDK.podspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name                  = 'RadarSDK'
-  s.version               = '3.8.7'
+  s.version               = '3.8.8-beta.2'
```
Comment (tjulien) [inline RadarSDK.podspec:3]: rm beta

File: RadarSDK/RadarUtils.m [hunk 3]
```diff
@@ -45,7 +45,7 @@ + (NSNumber *)timeZoneOffset {
 }
 
 + (NSString *)sdkVersion {
-    return @"3.8.7";
+    return @"3.8.8-beta.3";
```
Comment (tjulien) [inline RadarSDK/RadarUtils.m:48]: rm beta

File: RadarSDK/RadarTrackingOptions.m [hunk 4]
```diff
@@ -53,7 +53,7 @@ + (RadarTrackingOptions *)presetContinuous {
     options.startTrackingAfter = nil;
     options.stopTrackingAfter = nil;
     options.syncLocations = RadarTrackingOptionsSyncAll;
-    options.replay = RadarTrackingOptionsReplayNone;
+    options.replay = RadarTrackingOptionsReplayAll;
```
Comment (tjulien) [inline RadarSDK/RadarTrackingOptions.m:56]: revert these 3

File: RadarSDK/RadarDelegateHolder.m [hunk 5]
```diff
@@ -34,8 +34,8 @@ - (void)didReceiveEvents:(NSArray<RadarEvent *> *)events user:(RadarUser *)user
 
     for (RadarEvent *event in events) {
         [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo
-                                           message:[NSString stringWithFormat:@"📍 Radar event received | type = %@; link = https://radar.com/dashboard/events/%@",
-                                                                              [RadarEvent stringForType:event.type], event._id]];
+                                           message:[NSString stringWithFormat:@"📍 Radar event received | type = %@; replayed= %d link = https://radar.com/dashboard/events/%@",
```
Comment (tjulien) [inline RadarSDK/RadarDelegateHolder.m:37]: We should bring this change to Android for consistency

File: RadarSDK/RadarAPIClient.m [hunk 6]
```diff
@@ -138,6 +138,18 @@ - (void)flushReplays:(NSArray<NSDictionary *> *_Nonnull)replays
                            logPayload:NO
                       extendedTimeout:YES
                     completionHandler:^(RadarStatus status, NSDictionary *_Nullable res) {
+                            id eventsObj = res[@"events"];
+
+                            NSArray<RadarEvent *> *events = [RadarEvent eventsFromObject:eventsObj];
+                            if (events.count) {
+                                NSArray *eventsArr = (NSArray *)eventsObj;
+                                NSDictionary *lastEvent = [eventsArr lastObject];
+                                NSDictionary *userObj = lastEvent[@"user"];
```
Comment (tjulien) [inline RadarSDK/RadarAPIClient.m:147]: I think it's safer to keep it consistent with Android and read the user from the top level

Comment (lmeier) [reply RadarSDK/RadarAPIClient.m]: Oh yes, just didn't have this pushed up. https://github.com/radarlabs/radar-sdk-ios/pull/269/commits/8de0cb56f03163edcea54b3e8d82dcc02847179e

File: RadarSDK/RadarDelegateHolder.m (hunk 5 above)
Comment (lmeier) [reply RadarSDK/RadarDelegateHolder.m]: Great point, done. https://github.com/radarlabs/radar-sdk-android/pull/312/commits/8535265721ea18c2f7953be368d5afac748c01c1

File: RadarSDK/RadarDelegateHolder.m [hunk 7]
```diff
@@ -34,8 +34,8 @@ - (void)didReceiveEvents:(NSArray<RadarEvent *> *)events user:(RadarUser *)user
 
     for (RadarEvent *event in events) {
         [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo
-                                           message:[NSString stringWithFormat:@"📍 Radar event received | type = %@; link = https://radar.com/dashboard/events/%@",
-                                                                              [RadarEvent stringForType:event.type], event._id]];
+                                           message:[NSString stringWithFormat:@"📍 Radar event received | type = %@ replayed= %d; link = https://radar.com/dashboard/events/%@",
```
Comment (tjulien) [inline RadarSDK/RadarDelegateHolder.m:37]: I guess we need a semi colon in between type and replayed

**Review by tjulien — APPROVED** [review]

**Pull request merged**